include/zephyr/arch/arch_interface.h

```diff
@@ -46,6 +46,8 @@ typedef struct z_thread_stack_element k_thread_stack_t;
 
 typedef void (*k_thread_entry_t)(void *p1, void *p2, void *p3);
 
+typedef struct __esf z_arch_esf_t;
```

Comment:
Unfortunately, not all arch define its esf struct as `struct __esf`, this CI will fail and probably depends on #73593